mypy/checker.py

```diff
@@ -2336,6 +2336,29 @@ def check_assignment(self, lvalue: Lvalue, rvalue: Expression, infer_lvalue_type
                 self.infer_variable_type(inferred, lvalue, rvalue_type, rvalue)
             self.check_assignment_to_slots(lvalue)
 
+    def check_except_reachability(self, s: TryStmt) -> None:
+        """Perform except block reachability checks.
+
+        1. Check for duplicate exception clauses.
+        2. Check if super class has already been caught.
+        """
+        seen: List[Type] = []
+        for expr in s.types:
+            if expr and isinstance(expr, NameExpr) and isinstance(expr.node, TypeInfo):
+                with self.expr_checker.msg.disable_errors():
+                    typ = get_proper_type(self.check_except_handler_test(expr))
+                if isinstance(typ, AnyType) or isinstance(typ, UnionType):
```

Comment:
Sorry, this is not what I meant 🙂 

Theoretically we can have a case when some var has `type[SomeException] | Any` type. For example, from unresolved import or some runtime magic.

Is there anything we can do to handle this case here? Or shall we ignore it?

Reply:
I think I know what you mean. So, we do not have a variable, but a class whose type might be a Union. 

I do not how how to construct such a case for testing, can you help me with that? I was also not able to find something in `check-statements.test`

The tests already cover:
```
Parent: Any
class MyError(Parent): ...

omit: Union[Type[RuntimeError], Type[MyError]]
```

Nevertheless, it think this scenario would not pass the new `or isinstance(typ, UnionType)` filter, thus we would just ignore it.

Reply:
@sobolevn any updates here? ;)